apollo-ios/Sources/ApolloAPI/SchemaMetadata.swift

```diff
@@ -53,10 +53,22 @@ extension SchemaMetadata {
   /// - Returns: A `String` representing the cache key for the `object` to be used by
   /// `NormalizedCache` mechanisms.
   @inlinable public static func cacheKey(for object: ObjectData) -> String? {
-    guard let type = graphQLType(for: object),
-          let info = configuration.cacheKeyInfo(for: type, object: object) else {
-      return nil
+    guard let type = graphQLType(for: object) else { return nil }
+    
+    if let info = configuration.cacheKeyInfo(for: type, object: object) {
+      return "\(info.uniqueKeyGroup ?? type.typename):\(info.id)"
+    }
+    
+    guard let keyFields = type.keyFields else { return nil }
+    
+    let idValues = try? keyFields.map {
+      guard let v = object[$0] else {
+        throw JSONDecodingError.missingValue
```

Comment:
If we are allowing for nullable fields to be used, then we need to handle the case where those fields are null. I'm not sure if we should be or not.

Reply:
Non-nullable is now required.